src/AirshipBindings.NETStandard/AirshipBindings.NETStandard.Android/Airship.cs

```diff
@@ -82,7 +84,28 @@ public string NamedUser
                 }
             }
         }
-        
+
+        private InboxHandler onInboxUpdatedReceived;
+        public event InboxHandler OnInboxUpdatedReceived
+        {
+            add
+            {
+                onInboxUpdatedReceived += value;
+                UAirship.Shared().Inbox.RemoveListener(this);
```

Comment:
You should hoo, up the listener once when the instance is created. Use the shared with a callback to do so: 
```
UAirship.Shared(() => {
                UAirship.Shared().Inbox.AddListener(this);
})
```

Then I dont think you need to override the event handler.

Reply:
iOS you can do the same. Instead of trying to use the delegate sign up for an NSNotificaationEvent - https://github.com/urbanairship/ios-library-dev/blob/master/Airship/AirshipMessageCenter/Source/Inbox/UAInboxMessageList.m#L14